### ISSUE TITLE
unicorn/prefer-at: skip object numeric-key access (upstream #2999)

### ISSUE THREAD

**Issue by alexander7161** [oxc-project/oxc#23870]
## Summary

Upstream `eslint-plugin-unicorn` fixed [#2229](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2229) in [PR #2999](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2999) (merged 2026-06-05): `unicorn/prefer-at` with `checkAllIndexAccess: true` no longer reports or autofixes numeric-key lookup on plain objects.

Oxlint `1.71.0` still reports and autofixes these, producing broken code.

## Reproduction

Config:

```json
{
  "unicorn/prefer-at": ["error", { "checkAllIndexAccess": true }]
}
```

Input:

```ts
const tokens = { 60: '#666666', 10: '#F8F8F8' };
const color = tokens[60];
```

Run `oxlint --fix`.

**Actual:** autofixes to `tokens.at(60)` — objects have no `.at()`; TypeScript reports `TS2339`.

**Expected (upstream unicorn):** no diagnostic; leave `tokens[60]` unchanged.

## Upstream reference

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2229
- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2999

Please port the object-vs-array discrimination from upstream unicorn.

## Version

oxlint 1.71.0
